iis-wf/iis-wf-export-actionmanager/src/main/java/eu/dnetlib/iis/wf/export/actionmanager/module/DocumentSimilarityActionBuilderModuleFactory.java

```diff
@@ -103,60 +93,42 @@ public DocumentSimilarityActionBuilderModule(Float trustLevelThreshold, Float si
-            Oaf.Builder oafBuilder = Oaf.newBuilder();
-            oafBuilder.setKind(Kind.relation);
-            oafBuilder.setRel(buildOafRel(sourceId, targetDocId, score, invert));
-            oafBuilder.setDataInfo(buildInferenceForTrustLevel(StaticConfigurationProvider.ACTION_TRUST_0_9));
-            oafBuilder.setLastupdatetimestamp(System.currentTimeMillis());
-            return oafBuilder.build();
+        private Relation buildRelation(String sourceId, String targetDocId, float score, boolean invert) {
+            Relation relation = new Relation();
+            relation.setSource(invert?targetDocId:sourceId);
+            relation.setTarget(invert?sourceId:targetDocId);
+            relation.setRelType(REL_TYPE);
+            relation.setSubRelType(SUBREL_TYPE);
+            relation.setRelClass(invert ? REL_CLASS_IS_AMONG_TOP_N : REL_CLASS_HAS_AMONG_TOP_N);
+            relation.setDataInfo(buildInferenceForTrustLevel(StaticConfigurationProvider.ACTION_TRUST_0_9));
+            relation.setLastupdatetimestamp(System.currentTimeMillis());
+            throw new RuntimeException("awaiting Relation model extension required to export score");
+            // FIXME extend Relation model to export similarity score, replace throwing Exception with the two lines below
+            // relation.setProperties(Arrays.asList(buildSimilarityLevel(score)));
+            // return relation;
         }
-        
-        /**
-         * Builds {@link OafRel} object with similarity details.
-         */
-        private OafRel buildOafRel(String sourceId, String targetDocId, float score, boolean invert) {
-            OafRel.Builder relBuilder = OafRel.newBuilder();
-            relBuilder.setSource(invert?targetDocId:sourceId);
-            relBuilder.setTarget(invert?sourceId:targetDocId);
-            String relClass = invert ? Similarity.RelName.isAmongTopNSimilarDocuments.toString()
-                    : Similarity.RelName.hasAmongTopNSimilarDocuments.toString();
-            relBuilder.setChild(false);
-            relBuilder.setRelType(RelType.resultResult);
-            relBuilder.setSubRelType(SubRelType.similarity);
-            relBuilder.setRelClass(relClass);
-            ResultResult.Builder resultResultBuilder = ResultResult.newBuilder();
-            Similarity.Builder similarityBuilder = Similarity.newBuilder();
-            similarityBuilder.setRelMetadata(
-                    BuilderModuleHelper.buildRelMetadata(InfoSpaceConstants.SEMANTIC_SCHEME_DNET_RELATIONS_RESULT_RESULT, relClass));
-            similarityBuilder.setSimilarity(score);
-            similarityBuilder.setType(Type.STANDARD);
-            resultResultBuilder.setSimilarity(similarityBuilder.build());
-            relBuilder.setResultResult(resultResultBuilder.build());
-            return relBuilder.build();
+
+        private KeyValue buildSimilarityLevel(float score) {
```

Comment:
Yes, because the reference to this method will be uncommented once we get the updated Oaf model version with the `Relation` class supplemented with `properties` field (see the uncommented code few lines above).

I have prepared this code to make the upcoming commit as small as possible. by removing `//` from the lines above.

Reply:
Should've checked commented code  ;)